iOS/DuckDuckGo/Refactoring/Updated/ButtonConfigurations/ToolbarButton.swift

```diff
@@ -0,0 +1,123 @@
+
+    enum ButtonType {
+        case primary
+        case fire
+        case secondary
+    }
+
+    let type: ButtonType
+
+    init(_ type: ButtonType = .primary) {
+        self.type = type
+        super.init(frame: .zero)
+
+        applyConfiguration()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func setImage(_ image: UIImage?) {
+        configuration?.image = image
+    }
+
+    func applyConfiguration() {
+        configuration = .omniBarDefault()
+
+        let type = self.type
+
+        configuration?.automaticallyUpdateForSelection = false
+        configuration?.imageColorTransformer = .init { [weak self] _ in
+            type.foregroundColor(for: self?.state ?? .normal)
+        }
+
+        configurationUpdateHandler = { button in
+
+            var newConfiguration = button.configuration ?? .omniBarDefault()
+
+            newConfiguration.baseForegroundColor = type.foregroundColor(for: button.state)
+            newConfiguration.baseBackgroundColor = type.backgroundColor(for: button.state)
+
+            UIViewPropertyAnimator(duration: 0.25, curve: .easeInOut) {
+                button.configuration = newConfiguration
+            }.startAnimation()
+        }
+    }
+}
+
+private extension ToolbarButton.ButtonType {
+
+    func backgroundColor(for state: UIButton.State) -> UIColor {
+
+        switch (self, state) {
+        case (.fire, .highlighted):
+            return UIColor(designSystemColor: .controlsFillTertiary)
+        case (.fire, _):
+            return UIColor(designSystemColor: .controlsFillPrimary)
+        case (_, .highlighted):
+            return UIColor(designSystemColor: .controlsFillPrimary)
+        default:
+            return .clear
+        }
+    }
+
+    func foregroundColor(for state: UIButton.State) -> UIColor {
+
+        switch self {
+        case .primary, .fire:
+            switch state {
+            case .disabled:
+                return UIColor(designSystemColor: .icons).withAlphaComponent(0.5)
+            default:
+                return UIColor(designSystemColor: .icons)
+            }
+        case .secondary:
+            switch state {
+            case .disabled:
+                return UIColor(designSystemColor: .iconsSecondary).withAlphaComponent(0.5)
+            default:
+                return UIColor(designSystemColor: .iconsSecondary)
+            }
+        }
+    }
+}
+
+private extension UIButton.Configuration {
+    static func omniBarDefault() -> UIButton.Configuration {
+        var config = UIButton.Configuration.gray()
+        config.cornerStyle = .dynamic
+        config.buttonSize = .medium
+        config.titleAlignment = .center
+        config.background.backgroundInsets = .init(top: 2, leading: 2, bottom: 2, trailing: 2)
+
+        config.background.cornerRadius = 14
+
+//        config.contentInsets = .init(top: 8, leading: 16, bottom: 8, trailing: 16)
```

Comment:
Is it needed?

Reply:
It's not: https://github.com/duckduckgo/apple-browsers/pull/527/commits/43defc03063e2c1597c109655b89816f02a460ef